dcpy/extract/ingest.py

```diff
@@ -0,0 +1,33 @@
+from dcpy.connectors.edm import recipes
+from . import PARQUET_PATH, models
+
+
+# doesn't strictly need to be recipe config, but at least config-like object
+def archive_raw_dataset(dataset: str, version: str | None = None) -> recipes.Config:
+    """Given dataset and potentially a semantic version, ingest it from its source and archive it in its
+    raw format, with metadata, in edm-recipes/raw_data"""
+    import_definition = models.read_import_definition(dataset)
+
+    ## logic to grab dataset based on ImportDefinition, dump locally
+
+    ## logic to "compute" recipes.Config object, similar to Config.compute in library
```

Comment:
I'm leaning towards computing `Config` object separately and pass it into `archive_raw_dataset` because:
1) should make testing of `Config` object and archival of raw data easier as these functions will have 1 task to perform. 
2) `archive_raw_data` will be more readable.
3) It feels confusing getting a `Config` object as a result of a fn named `archive_raw_data`. Otherwise, I would rename the fn something like `archived_raw_data_config`

Let me know your thoughts

Reply:
That makes sense to me!

Reply:
See commit 4. Other thing in general that I would LOVE thoughts on is naming of things. I feel like I'm going in circles and using synonyms to mean different things and am just too deep in it at this point to know what makes sense

Reply:
Okay, will keep in mind to share feedback about naming! And I agree, "template" makes sense since we use it when we specifically refer THE templates lol